syntax = "proto3";

package WordVector;

service WordVector {
 rpc DetectLanguages (DetectLanguagesRequest) returns (DetectLanguagesReply) {}
 rpc GetVectors (GetVectorsRequest) returns (GetVectorsReply) {}
}

message DetectLanguagesRequest {
 repeated string texts = 1;
}

message DetectLanguagesReply {
 repeated DetectedLanguage results = 1;
}

message GetVectorsRequest {
 repeated string tokens = 1;
}

message GetVectorsReply {
 repeated Vector vectors = 1;
}

message DetectedLanguage {
 string text = 1;
 string language = 2;
}

message Vector {
 float frequency = 1;
 bytes data = 2;
}